src/sqllogictest/src/runner.rs

```diff
@@ -1739,6 +1901,298 @@ impl<'a> RewriteBuffer<'a> {
+    // the `SELECT` query, if any. Ordering is not preserved when a
+    // view includes an `ORDER BY` clause and must be re-enforced by
+    // an external `ORDER BY` clause when querying the view.
+    let (view_order_by, extra_columns, distinct) = if num_attributes.is_none() {
+        (query.order_by.clone(), vec![], None)
+    } else {
+        derive_order_by(&query.body, &query.order_by)
+    };
+
+    // Since one-shot SELECT statements may contain ambiguous column names,
+    // we either use the expected column names, if that option was
+    // provided, or else just rename the output schema of the view
+    // using numerically increasing attribute names, whenever possible.
+    // This strategy makes it possible to use `CREATE INDEX`, thus
+    // matching the behavior of the option `auto_index_tables`. However,
+    // we may be presented with a `SELECT *` query, in which case the parser
+    // does not produce sufficient information to allow us to compute
+    // the number of output columns. In the latter case, we are supplied
+    // with `None` for `num_attributes` and just employ the command
+    // `CREATE DEFAULT INDEX` instead. Additionally, the view is created
+    // without schema renaming. This strategy is insufficient to dodge
+    // column name ambiguity in all cases, but we assume here that we
+    // can adjust the (hopefully) small number of tests that eventually
+    // challenge us in this particular way.
+    let name = UnresolvedItemName(vec![Ident::new(format!("v{}", view_uuid))]);
+    let projection = expected_column_names.map_or(
+        num_attributes.map_or(vec![], |n| {
+            (1..=n).map(|i| Ident::new(format!("a{i}"))).collect()
+        }),
+        |cols| cols.iter().map(|c| Ident::new(c.as_str())).collect(),
+    );
+    let columns: Vec<Ident> = projection
+        .iter()
+        .cloned()
+        .chain(extra_columns.iter().map(|item| {
+            if let SelectItem::Expr {
+                expr: _,
+                alias: Some(ident),
+            } = item
+            {
+                ident.clone()
+            } else {
+                unreachable!("alias must be given for extra column")
+            }
+        }))
+        .collect();
+
+    // Build a `CREATE VIEW` with the columns computed above.
+    let mut query = query.clone();
+    if extra_columns.len() > 0 {
+        match &mut query.body {
+            SetExpr::Select(stmt) => stmt.projection.extend(extra_columns.iter().cloned()),
+            _ => unimplemented!("cannot yet rewrite projections of nested queries"),
+        }
+    }
+    let create_view = AstStatement::<Raw>::CreateView(CreateViewStatement {
+        if_exists: IfExistsBehavior::Error,
+        temporary: false,
+        definition: ViewDefinition {
+            name: name.clone(),
+            columns: columns.clone(),
+            query,
+        },
+    })
+    .to_ast_string_stable();
```

Comment:
I'm wondering if it would be helpful to print out the views/indexes created here too when `-v -v` is used.

Reply:
Good idea, I just added some support now to output the automatically generated view creation, indexing, querying, and dropping statements when the verbosity level is set high.